lib/auth/httpfallback.go

```diff
@@ -58,3 +61,120 @@ func (c *Client) RotateExternalCertAuthority(ctx context.Context, ca types.CertA
 
 	return trace.Wrap(err)
 }
+
+func (c *Client) deleteRemoteClusterLegacy(ctx context.Context, clusterName string) error {
+	if clusterName == "" {
+		return trace.BadParameter("missing parameter cluster name")
+	}
+	_, err := c.Delete(ctx, c.Endpoint("remoteclusters", clusterName))
+	return trace.Wrap(err)
+}
+
+// DeleteRemoteCluster deletes remote cluster by name
+// TODO(noah): DELETE IN 17.0.0
+func (c *Client) DeleteRemoteCluster(ctx context.Context, name string) error {
+	err := c.APIClient.DeleteRemoteCluster(ctx, name)
+	if err == nil {
+		return nil
+	}
+	if !trace.IsNotImplemented(err) {
+		return trace.Wrap(err)
+	}
+	return c.deleteRemoteClusterLegacy(ctx, name)
+}
+
+func (c *Client) getRemoteClustersLegacy(ctx context.Context) ([]types.RemoteCluster, error) {
+	out, err := c.Get(ctx, c.Endpoint("remoteclusters"), url.Values{})
+	if err != nil {
+		return nil, trace.Wrap(err)
+	}
+	var items []json.RawMessage
+	if err := json.Unmarshal(out.Bytes(), &items); err != nil {
+		return nil, trace.Wrap(err)
+	}
+	conns := make([]types.RemoteCluster, len(items))
+	for i, raw := range items {
+		conn, err := services.UnmarshalRemoteCluster(raw)
+		if err != nil {
+			return nil, trace.Wrap(err)
+		}
+		conns[i] = conn
+	}
+	return conns, nil
+}
+
+// GetRemoteClusters returns a list of remote clusters
+// Prefer using ListRemoteClusters.
+// TODO(noah): DELETE IN 17.0.0
+func (c *Client) GetRemoteClusters(ctx context.Context) ([]types.RemoteCluster, error) {
+	var rcs []types.RemoteCluster
+	pageToken := ""
+	for {
+		var err error
+		var page []types.RemoteCluster
+		page, pageToken, err = c.APIClient.ListRemoteClusters(ctx, 0, pageToken)
+		if err != nil {
+			if trace.IsNotImplemented(err) {
+				return c.getRemoteClustersLegacy(ctx)
+			}
+			return nil, trace.Wrap(err)
+		}
+		rcs = append(rcs, page...)
+		if pageToken == "" {
+			return rcs, nil
+		}
+	}
```

Comment:
This flows better I think
```suggestion
	pageToken := ""
	for {
		page, nextToken, err := c.APIClient.ListRemoteClusters(ctx, 0, pageToken)
		if err != nil {
			if trace.IsNotImplemented(err) {
				return c.getRemoteClustersLegacy(ctx)
			}
			return nil, trace.Wrap(err)
		}
		rcs = append(rcs, page...)
		if nextToken == "" {
			return rcs, nil
		}
		pageToken = nextToken
	}
```